src/cashier_frontend/src/services/link/link.service.ts

```diff
@@ -87,6 +87,15 @@ class LinkService {
 
     constructor(identity?: Identity | PartialIdentity | undefined) {
         const agent = HttpAgent.createSync({ identity, host: IC_HOST });
+        if (IS_LOCAL) {
+            agent.fetchRootKey().catch((err: Error) => {
```

Comment:
This logic is duplicated in at least 6 different places. It should be extracted to a common module.
In addition, `IS_LOCAL` usage is abused. We need instead speficic feature flags. For example `FEATURE_FLAGS_ENABLE_LOCAL_IDENTITY_PROVIDER`, etc.